docs/api.rst

```diff
@@ -18,6 +18,7 @@ Top-Level
    set_backend
    readxml
    writexml
+   compat
```

Comment:
@lukasheinrich `pyhf.compat` needs to be added to the public API docs. I've added it to top-level, but please move it around as you think best. :+1: